libraries/griptape_nodes_library/griptape_nodes_library/video/wan_animate_generation.py

```
@@ -0,0 +1,556 @@
+
+        return generation_id
+
+    def _build_payload(self, params: dict[str, Any]) -> dict[str, Any]:
+        # Build payload matching proxy expected format
+        payload = {
+            "input": {
+                "image_url": params["image_url"],
+                "video_url": params["video_url"],
+            },
+            "parameters": {
+                "mode": params["mode"],
+                "duration": params["duration"],
+            },
+        }
+
+        return payload
+
+    def _poll_for_result(self, generation_id: str, headers: dict[str, str], video_duration: int) -> None:
+        get_url = urljoin(self._proxy_base, f"generations/{generation_id}")
+        start_time = monotonic()
+        last_json = None
+        attempt = 0
+        poll_interval_s = 5.0
+        timeout_s = video_duration * 20.0
+
+        while True:
+            if monotonic() - start_time > timeout_s:
+                self.parameter_output_values["video"] = self._extract_video_url(last_json)
+                logger.debug("Polling timed out waiting for result")
+                self._set_status_results(
+                    was_successful=False,
+                    result_details=f"Video generation timed out after {timeout_s} seconds waiting for result.",
+                )
+                return
+
+            try:
+                get_resp = requests.get(get_url, headers=headers, timeout=60)
+                get_resp.raise_for_status()
+                last_json = get_resp.json()
+                self.parameter_output_values["provider_response"] = last_json
+            except Exception as exc:
+                logger.debug("GET generation failed: %s", exc)
+                error_msg = f"Failed to poll generation status: {exc}"
+                self._set_status_results(was_successful=False, result_details=error_msg)
+                self._handle_failure_exception(RuntimeError(error_msg))
+                return
+
+            with suppress(Exception):
```

Why are we suppressing exceptions for a log?